src/components/library/library.jsx

```diff
@@ -172,33 +164,28 @@ class LibraryComponent extends React.Component {
                     })}
                     ref={this.setFilteredDataRef}
                 >
-                    {this.getFilteredData().map((dataItem, index) => {
-                        const scratchURL = dataItem.md5 ?
-                            `https://cdn.assets.scratch.mit.edu/internalapi/asset/${dataItem.md5}/get/` :
-                            dataItem.rawURL;
-                        return (
-                            <LibraryItem
-                                bluetoothRequired={dataItem.bluetoothRequired}
-                                collaborator={dataItem.collaborator}
-                                description={dataItem.description}
-                                disabled={dataItem.disabled}
-                                extensionId={dataItem.extensionId}
-                                featured={dataItem.featured}
-                                hidden={dataItem.hidden}
-                                iconURL={scratchURL}
-                                id={index}
-                                insetIconURL={dataItem.insetIconURL}
-                                internetConnectionRequired={dataItem.internetConnectionRequired}
-                                key={`item_${index}`}
-                                name={dataItem.name}
-                                onBlur={this.handleBlur}
-                                onFocus={this.handleFocus}
-                                onMouseEnter={this.handleMouseEnter}
-                                onMouseLeave={this.handleMouseLeave}
-                                onSelect={this.handleSelect}
-                            />
-                        );
-                    })}
+                    {this.getFilteredData().map((dataItem, index) => (
+                        <LibraryItem
+                            bluetoothRequired={dataItem.bluetoothRequired}
+                            collaborator={dataItem.collaborator}
+                            description={dataItem.description}
+                            disabled={dataItem.disabled}
+                            extensionId={dataItem.extensionId}
+                            featured={dataItem.featured}
+                            hidden={dataItem.hidden}
+                            iconMd5={dataItem.md5}
+                            iconRawURL={dataItem.rawURL}
+                            icons={dataItem.json && dataItem.json.costumes}
```

Comment:
it does decide for itself; but rather than passing the entire object, i'm passing only the relevant fields. Alternatively, I could pass the whole library item... should I try that?

Reply:
Hmm... I feel strongly both ways. :/

There's a part of me that really feels like there should be less distinction between a single-icon item and one that rotates through multiple costumes -- that a non-animated item should be the same as an animated item with only one costume. That said, I'm not sure it's worth the extra data-mangling to make that true just because it feels more pretty.

Another thing I'm thinking about is the code I have in my scratch-desktop branch of scratch-gui, which (among other changes) puts MD5 and rawURL into a unified "imageSource" object. I'm trying not to be too distracted from my principles by the idea of a rough merge ;)

My inclination is to say that this seems good for now since this is a big step toward "better" and I'm probably going to need to make minor changes anyway as I merge this with my scratch-desktop stuff. At the moment I'm not sure what the shape of those changes will be, but we can debate them in the PR I'll submit around January 7 or so :)

All that said, I'm 120% open to suggestion on how this should end up, and @benjiwheeler I certainly don't want to step on your toes if you'd like to make changes.